kroxylicious-filters/kroxylicious-record-validation/src/main/java/io/kroxylicious/proxy/filter/schema/validation/bytebuf/JsonSchemaBytebufValidator.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.filter.schema.validation.bytebuf;
+
+import java.nio.ByteBuffer;
+import java.util.Map;
+import java.util.Optional;
+
+import org.apache.kafka.common.record.Record;
+
+import io.apicurio.registry.resolver.strategy.ArtifactReference;
+import io.apicurio.schema.validation.json.JsonValidationResult;
+import io.apicurio.schema.validation.json.JsonValidator;
+
+import io.kroxylicious.proxy.filter.schema.validation.Result;
+
+public class JsonSchemaBytebufValidator implements BytebufValidator {
+
+    private final JsonValidator jsonValidator;
+
+    public JsonSchemaBytebufValidator(Map<String, Object> schemaResolverConfig, Long globalId) {
+        this.jsonValidator = new JsonValidator(schemaResolverConfig, Optional.of(ArtifactReference.fromGlobalId(globalId)));
+    }
+
+    @Override
+    public Result validate(ByteBuffer buffer, int length, Record record, boolean isKey) {
+        JsonValidationResult jsonValidationResult = jsonValidator.validateByArtifactReference(buffer);
```

Comment:
Yes, this is putting the schema registry in the path for every request. That said, as you correctly anticipated, the underlying library does support both caching schemas (by globalId or other strategies) and retries and they're already included inside the existing request. Right now, if the schema is not retrievable from the Schema Registry (after retries etc), it will manifest success == false.